File: backend/src/main/java/com/elms/service/LeaveBalanceService.java
package com.elms.service;

import com.elms.dto.LeaveBalanceDto;

import java.util.List;

public interface LeaveBalanceService {
    
    List<LeaveBalanceDto> getAllLeaveBalances();
    
    List<LeaveBalanceDto> getUserLeaveBalances(Long userId);
    
    LeaveBalanceDto getUserLeaveBalance(Long userId, Long leaveTypeId);
    
    LeaveBalanceDto createLeaveBalance(Long userId, Long leaveTypeId, Integer days);
    
    LeaveBalanceDto updateLeaveBalance(Long userId, Long leaveTypeId, Integer days);
    
    void deleteLeaveBalance(Long userId, Long leaveTypeId);
    
    void deductLeaveBalance(Long userId, Long leaveTypeId, Integer days);
}

File: frontend/src/services/leaveBalanceService.js
import api from './api';

export const leaveBalanceService = {
  getAllLeaveBalances: async () => {
    const response = await api.get('/leave-balances');
    return response.data;
  },

  getUserLeaveBalances: async (userId) => {
    const response = await api.get(`/leave-balances/user/${userId}`);
    return response.data;
  },

  getUserLeaveBalance: async (userId, leaveTypeId) => {
    const response = await api.get(`/leave-balances/user/${userId}/leave-type/${leaveTypeId}`);
    return response.data;
  },

  createLeaveBalance: async (userId, leaveTypeId, days) => {
    const response = await api.post('/leave-balances', null, {
      params: { userId, leaveTypeId, days },
    });
    return response.data;
  },

  updateLeaveBalance: async (userId, leaveTypeId, days) => {
    const response = await api.put(`/leave-balances/user/${userId}/leave-type/${leaveTypeId}`, null, {
      params: { days },
    });
    return response.data;
  },

  deleteLeaveBalance: async (userId, leaveTypeId) => {
    const response = await api.delete(`/leave-balances/user/${userId}/leave-type/${leaveTypeId}`);
    return response.data;
  },
};
